enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+* A [Grafana Operator](https://operatorhub.io/operator/grafana-operator)
+
+Once a `MonitoringStack` has been created, the operator will deploy the appropriate resources in the `MonitoringStackOperor`'s namespace based on how the stack is configured. In a simple case, this might be:
+
+* a Prometheus CR
+* a Grafana CR and a Grafana Data Source CR
+* a Thanos Querier if the stack is configured for high availability
+
+In other cases, it might be just a Prometheus Agent deployment (forwarding only capabilities).
+
+The `MonitoringStack` CRD is meant to be an abstraction over monitoring needs and requirements for a single tenant that specifies their needs. The underlying implementation is up to the MonitoringStackOperator.
+
+In this manner, managed service owners can simply create an instance of `MonitoringStack` alongside their services and delegate the provisioning and management of their monitoring stack to the operator.
+
+An example `MonitoringStack` resource could look as follows:
+
+```yaml
+---
+apiVersion: monitoring.coreos.io/v1alpha1
+kind: MonitoringStack
+metadata:
+    name: monitoring
+    namespace: redhat-kafka
+spec:
+    retention: 12d
+    resources:
+      requests: 
+        memory: 512M
+      limits:  
+        memory: 1G
+---
+```
+
+In the first iteration, each monitoring stack will be deployed and scoped to a single namespace. This means that the Prometheus instance will only be able to scrape monitoring resources from the namespace it is deployed in. In subsequent iterations, the CRD can be extended to allow scraping from multiple namespaces, either by namespace name or by namespace label selector.
+
+With such a design, User Workload Monitoring could become just another instance of a `MonitoringStack`. Furthermore, Platform Monitoring can be trimmed down in the same way by allowing various OpenShift core components to deploy their `MonitoringStack` resource.
+
+The following diagram illustrates a high-level overview of the final solution applied in the context of managed services.
+
+![drawing](https://docs.google.com/drawings/d/e/2PACX-1vT4p10FDLEg0bv_pLUlwyVszq1275MNRJ-sBZMLxT5gcZ0q-tY8MHoEYpOlVmJuxlT40lqUQYCDwaSC/pub?w=1213&h=898)
+
+_First iteration of the new monitoring operator, showing a monitoring stack with a Prometheus, Grafana and a shared Alertmanager._
+
+We recommend using either [OperatorSDK](https://github.com/operator-framework/operator-sdk) or [KubeBuilder](https://github.com/kubernetes-sigs/kubebuilder) to develop the operator to increase development efficiency by taking advantage of the higher abstraction constructs, the rich documentation, and a vibrant community.
+Alternatively, we could also consider using a higher-level operator, such as [locutus](https://github.com/brancz/locutus), for rapid development with a path for GitOps deployments. In any case, this is an implementation detail that should be left to the team implementing this to decide on.
```

Comment:
Nit: I'd love to see us being consistent with other Addons/RH-operators here if possible. 

New bundle/delivery strategies sound like a small change but require training a lot of people across several orgs.